lib/beaker/host/unix/exec.rb

```diff
@@ -278,6 +278,8 @@ def clear_env_var key
   # @return [Result] result of restarting the SSH service
   def ssh_service_restart
     case self['platform']
+    when /ubuntu-24.04-/
+      exec(Beaker::Command.new("systemctl restart ssh"))
```

Comment:
did they actually name the unit `ssh.service` and not `sshd.service` anymore?

Reply:
Yes it seems they have and sshd.service throws error.

Reply:
Indeed `ssh.service` is the sshd server. As for why Ubuntu would rename the service, I have no idea

```
# cat  /usr/lib/systemd/system/ssh.service
[Unit]
Description=OpenBSD Secure Shell server
Documentation=man:sshd(8) man:sshd_config(5)
After=network.target auditd.service
ConditionPathExists=!/etc/ssh/sshd_not_to_be_run

[Service]
EnvironmentFile=-/etc/default/ssh
ExecStartPre=/usr/sbin/sshd -t
ExecStart=/usr/sbin/sshd -D $SSHD_OPTS
ExecReload=/usr/sbin/sshd -t
ExecReload=/bin/kill -HUP $MAINPID
KillMode=process
Restart=on-failure
RestartPreventExitStatus=255
Type=notify
RuntimeDirectory=sshd
RuntimeDirectoryMode=0755

[Install]
WantedBy=multi-user.target
Alias=sshd.service
```

Reply:
but if it has `Alias=sshd.service` it should be possible to run `systemctl restart sshd.service`?

Reply:
Hmm, note this line `Alias=sshd.service`. For some reason earlier versions of ubuntu allow the service to managed via `ssh` or `sshd`. For example, Ubuntu 22.04 accepts either:

```
root@forlorn-field:~# systemctl status sshd | head -2
● ssh.service - OpenBSD Secure Shell server
     Loaded: loaded (/lib/systemd/system/ssh.service; enabled; vendor preset: enabled)
root@forlorn-field:~# systemctl status ssh | head -2
● ssh.service - OpenBSD Secure Shell server
     Loaded: loaded (/lib/systemd/system/ssh.service; enabled; vendor preset: enabled)
```

Perhaps this is a bug in our 24.04 images?

Reply:
This appears to be a regression in the version of systemd that ships in noble https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/2057768 Since "ssh" is the correct name (whereas "sshd" is the alias) and noble is a new platform I think this PR is fine.

Reply:
I am a bit torn between using this for every Ubuntu+Debian vs using it only as a workaround until the systemd bug is fixed. I tend to the latter because it keeps our code shorter

Reply:
Can we have check for just Ubuntu-24 and move forward with Ubuntu 24 support as this is blocking further development and fix this later?
@joshcooper @bastelfreak